glib/src/unichar.rs

```diff
@@ -0,0 +1,237 @@
+use crate::{
+    translate::{FromGlib, IntoGlib},
+    UnicodeBreakType, UnicodeScript, UnicodeType,
+};
+
+mod sealed {
+    pub trait Sealed {}
+    impl Sealed for char {}
+}
+
+// rustdoc-stripper-ignore-next
+/// The kind of decomposition to perform
+#[derive(Clone, Copy)]
+pub enum DecompositionKind {
+    // rustdoc-stripper-ignore-next
+    /// Compatbility decomposition
+    Compatibility,
+
+    // rustdoc-stripper-ignore-next
+    /// Canonical decomposition
+    Canonical,
+}
+
+impl DecompositionKind {
+    fn into_bool(self) -> bool {
+        use DecompositionKind::*;
+        match self {
+            Compatibility => true,
+            Canonical => false,
+        }
+    }
+}
+
+// rustdoc-stripper-ignore-next
+/// The result of a single step of the Unicode canonical decomposition algorithm
+pub enum DecompositionResult {
```

Comment:
I'm not too happy with the names, but I don't have better ideas

Reply:
Maybe `CharacterDecomposition`?

Reply:
Yeah I liked that name more